Formula/beast.rb

```diff
@@ -24,12 +24,18 @@ class Beast < Formula
   depends_on "openjdk@11"
 
   def install
-    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_prefix
+    ENV["JAVA_HOME"] = Language::Java.java_home("11")
     system "ant", "linux"
     libexec.install Dir["release/Linux/BEASTv*/*"]
     pkgshare.install_symlink libexec/"examples"
     bin.install Dir[libexec/"bin/*"]
-    bin.env_script_all_files libexec/"bin", JAVA_HOME: ENV["JAVA_HOME"]
+
+    env = Language::Java.overridable_java_home_env("11")
+    on_linux do
+      env["PATH"] = "$JAVA_HOME/bin:$PATH"
+    end
```

Comment:
Haven't check much on Ruby hashes, but am wondering if this could possibly fail on ordering.
The resulting script needs to be in the order:
```
JAVA_HOME=${JAVA_HOME:-...} PATH="$JAVA_HOME/bin:$PATH /path/to/libexec/bin/<command>
```
i.e. `JAVA_HOME` has to be assigned before it is used in `PATH`.

Reply:
@iMichka I'm actually a bit worried about the ordering problem. May need to re-consider.

Reply:
It does seem order of enumeration is guaranteed https://ruby-doc.org/core-2.6/Hash.html
> Hashes enumerate their values in the order that the corresponding keys were inserted.

So, I guess it should be fine if we are accessing the env vars this way.